src/modules/UI/scenes/ChangeMiningFee/ChangeMiningFee.ui.js

```diff
@@ -0,0 +1,69 @@
+import React, {Component} from 'react'
+import {View, Text, TextInput} from 'react-native'
+
+import RadioButton from './components/RadioButton.ui'
+import * as FEE from '../../../../constants/FeeConstants'
+import strings from '../../../../locales/default'
+
+import style from './style'
+
+const feeOptions = [
+  { value: FEE.LOW_FEE, label: 'mining_fee_low_label_choice' },
+  { value: FEE.STANDARD_FEE, label: 'mining_fee_standard_label_choice' },
+  { value: FEE.HIGH_FEE, label: 'mining_fee_high_label_choice' },
+  { value: FEE.CUSTOM_FEE, label: 'mining_fee_custom_label_choice' },
+]
+
+export default class ChangeMiningFee extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      fee: props.fee,
+      feeSetting: props.feeSetting,
+    }
+  }
+
+  componentWillUnmount () {
+    if (this.state.fee !== this.props.fee || this.state.feeSetting !== this.props.feeSetting) {
+      this.props.onSubmit(this.state.feeSetting, this.state.fee)
+    }
+  }
+
+  handlePress = (feeSetting) => this.setState({ feeSetting, fee: '' });
+  handleChange = (fee) => this.setState({ fee });
```

Comment:
Please filter fees to be numeric strings only, e.g `"123"` = good, `"123[;.sdfv"` = bad
One possible way of doing that:
```
const regex = /\D/g
const filteredFee = fee.replace(/\D/g, '')
this.setState({ fee: filteredFee })
```